source/common/statistic_impl.cc

```diff
@@ -236,6 +246,31 @@ nighthawk::client::Statistic HdrStatistic::toProto(SerializationDomain domain) c
   return proto;
 }
 
+absl::StatusOr<std::unique_ptr<std::istream>> HdrStatistic::serializeNative() const {
+  char* data;
+  if (hdr_log_encode(histogram_, &data) == 0) {
+    absl::string_view s(data, strlen(data));
+    auto ss = std::make_unique<std::stringstream>();
+    *ss << s;
+    free(data);
+    return ss;
+  }
+  ENVOY_LOG(error, "Failed to write HdrHistogram data.");
+  return absl::Status(absl::StatusCode::kInternal, "Failed to write HdrHistogram data");
+}
+
+absl::Status HdrStatistic::deserializeNative(std::istream& stream) {
+  std::string s(std::istreambuf_iterator<char>(stream), {});
+  struct hdr_histogram* new_histogram = nullptr;
```

Comment:
Can you help me understand the memory here? Who allocates and owns (eventually frees) the memory pointed o by `new_histogram`?

Reply:
Added some comments in 79bad46 to help with that.